pkg/graph/parser/parser.go

```diff
@@ -80,40 +80,45 @@ func validateSchema(schema *spec.Schema, path string) error {
 	return nil
 }
 
-func getExpectedType(schema *spec.Schema, resource interface{}) (string, error) {
+func getExpectedTypes(schema *spec.Schema, resource interface{}) ([]string, error) {
 	// handle "x-kubernetes-int-or-string"
 	if ext, ok := schema.VendorExtensible.Extensions[xKubernetesIntOrString]; ok {
 		enabled, ok := ext.(bool)
 		if !ok {
-			return "", fmt.Errorf("xKubernetesIntOrString extension is not a boolean")
+			return []string{""}, fmt.Errorf("xKubernetesIntOrString extension is not a boolean")
 		}
 		if enabled {
-			switch resourceType := resource.(type) {
-			case string:
-				return "string", nil
-			case int:
-				return "integer", nil
-			default:
-				return "", fmt.Errorf("found `xKubernetesIntOrString` extension but field value is neither a string nor an integer: %v", resourceType)
-			}
+			return []string{"string", "integer"}, nil
+		}
+		if !enabled {
+			return []string{""}, fmt.Errorf("found `xKubernetesIntOrString` extension but field value is neither a string nor an integer: %T", resource)
```

Comment:
Do we want to error if a CRD has `x-kubernetes-int-or-string: false`? seems like a case where we shouldn't error